### PR TITLE
777 Implemented feature for export student class book in html format

### LINKED ISSUE #777: Export student classbook to html
Implement feature for export student class book in html format (/api/v2/exports/studentClassbook/{studentId}/{extension})

### PR DESCRIPTION
**Pull request by BobTheSuperman**

### CODE REVIEW

File: CharlieBackend.Business/Services/FileServices/ExportFileServices/Html/StudentClassbookExportHTML.cs [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+namespace CharlieBackend.Business.Services.FileServices.ExportFileServices.Html
+{
+    class StudentClassbookExportHTML : BaseFileExportHTML
+    {
+        #region private constants
+        private const int _CourseColumnNumber = 0;
+        private const int _StudentGroupColumnNumber = 1;
+        private const int _DateColumnNumber = 2;
+        private const int _FourthColumn = 3;
+        private const int _FifthColumn = 4;
```
Comment (BobTheSuperman) [inline CharlieBackend.Business/Services/FileServices/ExportFileServices/Html/StudentClassbookExportHTML.cs:17]: Enum

File: CharlieBackend.Business/Services/FileServices/ExportFileServices/Html/StudentClassbookExportHTML.cs [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+                string[][] rows = new string[data.StudentsMarks.Count()][];
+
+                for (int i = 0; i < studentMarks.Count(); i++)
+                {
+                    rows[i] = new string[headers.Length];
+
+                    rows[i][_CourseColumnNumber] = studentMarks.ElementAt(i).Course;
+                    rows[i][_StudentGroupColumnNumber] = studentMarks.ElementAt(i).StudentGroup;
+                    rows[i][_DateColumnNumber] = studentMarks.ElementAt(i).LessonDate.ToString();
+                    rows[i][_FourthColumn] = studentVisits.ElementAt(i).Presence == null ? " " : studentVisits.ElementAt(i).Presence == true ? $"&#10003;" : " ";
```
Comment (BobTheSuperman) [inline CharlieBackend.Business/Services/FileServices/ExportFileServices/Html/StudentClassbookExportHTML.cs]: ✓(&#10003;) add to html constants

File: CharlieBackend.Business/Services/FileServices/ExportFileServices/Html/StudentClassbookExportHTML.cs [hunk 3]
```diff
@@ -0,0 +1,143 @@
...
+                finishDate = data.StudentsPresences.Last().LessonDate?.ToString("dd.MM.yyyy");
+            }
+            else if (data.StudentsMarks != null && data.StudentsMarks.Any())
+            {
+                student = data.StudentsMarks.First().Student;
+                startDate = data.StudentsMarks.First().LessonDate?.ToString("dd.MM.yyyy");
+                finishDate = data.StudentsMarks.Last().LessonDate?.ToString("dd.MM.yyyy");
+            }
+
+            return $"Student {student} results {startDate}{HtmlGenerator.NonBreakingSpace}{finishDate}";
```
Comment (ArsenMaistryshyn) [inline CharlieBackend.Business/Services/FileServices/ExportFileServices/Html/StudentClassbookExportHTML.cs]: please move to constants

File: CharlieBackend.Business/Services/FileServices/ExportFileServices/Html/StudentClassbookExportHTML.cs (hunk 2 above)
Comment (ArsenMaistryshyn) [inline CharlieBackend.Business/Services/FileServices/ExportFileServices/Html/StudentClassbookExportHTML.cs]: please format

**Review by ArsenMaistryshyn — CHANGES_REQUESTED** [review]

**Review by Ssashaa — APPROVED** [review]

File: CharlieBackend.Business/Services/FileServices/ExportFileServices/Html/StudentClassbookExportHTML.cs (hunk 2 above)
Comment (BobTheSuperman) [reply CharlieBackend.Business/Services/FileServices/ExportFileServices/Html/StudentClassbookExportHTML.cs]: Done, thx.

File: CharlieBackend.Business/Services/FileServices/ExportFileServices/Html/StudentClassbookExportHTML.cs [hunk 4]
```diff
@@ -1,20 +1,22 @@
...
 using System.Linq;
 using System.Text;
-using System.Threading.Tasks;
+using CharlieBackend.Business.Helpers;
+using CharlieBackend.Core.DTO.Dashboard;
 
 namespace CharlieBackend.Business.Services.FileServices.ExportFileServices.Html
 {
-    class StudentClassbookExportHTML : BaseFileExportHTML
+    internal class StudentClassbookExportHTML : BaseFileExportHTML
```
Comment (ArsenMaistryshyn) [inline CharlieBackend.Business/Services/FileServices/ExportFileServices/Html/StudentClassbookExportHTML.cs]: please make public

**Review by ArsenMaistryshyn — CHANGES_REQUESTED** [review]

**Review by da-real-ivanpan — APPROVED** [review]

**Review by ArsenMaistryshyn — APPROVED** [review]

**Pull request merged**